preprocess.py

```diff
@@ -11,25 +11,42 @@
     CustomLMDataset,
     get_collate_fn,
 )
+from cezo_fl.fl_helpers import get_client_name
 from datasets import load_dataset
 from transformers import AutoTokenizer
 
 
 def use_device(args):
+    num_clients = args.num_clients
+
     use_cuda = not args.no_cuda and torch.cuda.is_available()
     use_mps = not args.no_mps and torch.backends.mps.is_available()
     if use_cuda:
-        print("----- Using cuda -----")
+        num_gpu = torch.cuda.device_count()
+        print(f"----- Using cuda count: {num_gpu} -----")
         # num_workers will make dataloader very slow especially when number clients is large
         # Do not shuffle shakespeare
-        kwargs = {"pin_memory": True, "shuffle": args.dataset != "shakespeare"} if use_cuda else {}
-        return torch.device("cuda"), kwargs
+        kwargs = {"pin_memory": True, "shuffle": args.dataset != "shakespeare"}
+
+        return {
+            "server": torch.device("cuda:0"),
+            **{
+                get_client_name(i): torch.device(f"cuda:{(i+1) % num_gpu}")
+                for i in range(num_clients)
+            },
+        }, kwargs
```

Comment:
what is our python version? Now this can be 
```python
{"server": torch.device("cuda:0")} | {
       get_client_name(i): torch.device(f"cuda:{(i+1) % num_gpu}")
        for i in range(num_clients)
}
```
Or more readable with `server_device | client_devices` if define dict variable before.